functions/src/modules/SQR.functions.ts

```diff
@@ -205,3 +206,87 @@ export const processSubmissions = functions.database.ref('/webforms/sqr/{submiss
     
     return 1;
 });
+
+
+/////////////////////////////////////////////////
+//          Update Email Templates on sendInBlue 
+//          in response to changes of the templates
+//          on FB (Storage Triggered)
+//
+/////////////////////////////////////////////////
+
+export const updateEmailTemplates = functions.storage.object()
+.onFinalize(async object => {
+
+    const path = require('path');
+    const os = require('os');
+    const fs = require('fs');
+
+    const filePath = object.name;
+    // slice(0, -5) to get rid of the trailing '.html' extension
+    const fileName = filePath.split('/')[2].slice(0, -5);
+    const tempLocalFile = path.join(os.tmpdir(), fileName);
+    
+    
+    if (!filePath.startsWith('email/templates')) 
+        return 1;
+
+    let emailRef = bucket.file(object.name);
+    await emailRef.download({ destination: tempLocalFile });
```

Comment:
Maybe we can use a public URL of the file and pass it to the SendInBlue API rather than downloading it? Just consider, I'm not insisting.

Reply:
I tried it at first but it gave me a permission denied error `SigningError: Permission iam.serviceAccounts.signBlob is required to perform this operation on service account`
and I kinda got lost trying to find an answer on how to do it without a service account.
If you find it important I can retry and search again.

Reply:
Let's keep it for now.